miqa/core/tests/pyppeteer/test_import_export.py

```diff
@@ -0,0 +1,100 @@
+import pytest
+
+
+async def get_current_num_scans(page):
+    return len(await page.xpath('//ul[contains(@class, "scans")]/li'))
+
+
+async def edit_import_export_paths(page):
+    # Edit import path
+    import_path_input = await page.waitForXPath('//input[contains(@name, "import-path")]')
+    await import_path_input.click({'clickCount': 3})  # highlight the text to overwrite it
+    await import_path_input.type('samples/demo_project.csv')
+    await page.waitFor(1_000)
+
+    # Edit export path
+    import_path_input = await page.waitForXPath('//input[contains(@name, "export-path")]')
+    await import_path_input.click({'clickCount': 3})  # highlight the text to overwrite it
+    await import_path_input.type('samples/demo_project_export.csv')
+    await page.waitFor(1_000)
+
+    # Click save
+    await (await page.waitForXPath('//button[span[.=" Save "]][@type="submit"]')).click()
+    await page.waitFor(1_000)
+
+
+async def perform_import(page):
+    # Click import button, and again in the modal confirmation
+    await (await page.waitForXPath('//button[span[.=" Import "]]')).click()
+    await page.waitFor(1_000)
+    await (
+        await page.waitForXPath(
+            '//div[contains(@class, "v-card__actions")]//button[span[.=" Import "]]'
+        )
+    ).click()
+    await page.waitFor(6_000)
+    await page.waitForXPath('//div[@class="v-snack__content"][contains(., "Import finished")]')
+    # Wait for the snackbar message to go away
+    await page.waitFor(5_000)
+
+
+async def perform_export(page):
+    # Click export button
+    await (await page.waitForXPath('//button[span[.=" Export "]]')).click()
+    await page.waitFor(2_000)
+    await page.waitForXPath(
+        '//div[@class="v-snack__content"][contains(., "Saved data to file successfully")]'
+    )
+    # Wait for the snackbar message to go away
+    await page.waitFor(5_000)
+
+
+@pytest.mark.pyppeteer
+async def test_project_import_export(
+    page,
+    log_in,
+    webpack_server,
+    user_factory,
+    project_factory,
+):
+    creator = user_factory()
+    project = project_factory(creator=creator, name='Demo Project')
+    project.update_group('tier_2_reviewer', [creator])
+
+    await log_in(creator)
+    # Click on the project in the project selector
+    await (
+        await page.waitForXPath(f'//div[contains(@class, "col")][contains(.,"{project.name}")]')
+    ).click()
+    await page.waitFor(1_000)
```

Comment:
For future reference:
If you end up writing enough end to end tests that you have multiple tests that use the same page/widget to test completely different functionality, it becomes worth it to create modules/classes that represent a page/component in the app and keep all the helpers on that page helper rather than in the test method.

Setting that up now to abstract away 4 lines (2 statements?) of code seems like overkill, but it's something to think about planning for as you write more tests.